src/utils.ts

```diff
@@ -82,3 +82,17 @@ export function isFunctionalComponent(component: any) {
 export function isObjectComponent(component: any) {
   return typeof component !== 'function' && !isClassComponent(component)
 }
+
+// https://stackoverflow.com/questions/15458876/check-if-a-string-is-html-or-not/15458987#answer-15458968
+export function isHTML(str: string) {
+  var a = document.createElement('div')
+  a.innerHTML = str
+
+  for (let c = a.childNodes, i = c.length; i--; ) {
+    if (c[i].nodeType == 1) {
+      return true
+    }
+  }
+
+  return false
```

Comment:
OOC: ain't that a very expensive check that'll slow down the tests?

Reply:
I don't think so - it's the last case in the `reduce` function, and we return true early - so this only executes if you are passing a text slot, which is not super common from what I can tell.

I don't expect a *significant* performance impact. I have no idea how expensive this actually is, we might want to start having some automated benchmarking at some point.

Do you know of a cheaper way to do this? We could do something like check for both `<` and `>` in the string or something - that would likely solve for 99% of cases. Thoughts?

Reply:
You' re probably right, I missed that it was the last case, so it shouldn't matter much 👍.